src/renderer/pages/Onboarding/WalletConnect/NovaWallet.tsx

```diff
@@ -0,0 +1,144 @@
+import QRCodeStyling from 'qr-code-styling';
+import { useEffect, useRef, useState } from 'react';
+import { useUnit } from 'effector-react';
+
+import { useI18n } from '@renderer/app/providers';
+import { BaseModal, Button, HeaderTitleText, SmallTitleText } from '@renderer/shared/ui';
+import { Animation } from '@renderer/shared/ui/Animation/Animation';
+import ManageStep from './ManageStep/ManageStep';
+import novawallet_onboarding_tutorial from '@video/novawallet_onboarding_tutorial.mp4';
+import novawallet_onboarding_tutorial_webm from '@video/novawallet_onboarding_tutorial.webm';
+import { usePrevious } from '@renderer/shared/lib/hooks';
+import { getWalletConnectChains, wcModel } from '@renderer/entities/walletConnect';
+import { chainsService } from '@renderer/entities/network';
+import { wcOnboardingModel } from '@renderer/pages/Onboarding/WalletConnect/model/wc-onboarding-model';
+import { NWQRConfig, Step } from './common/const';
+import { useStatusContext } from '@renderer/app/providers/context/StatusContext';
+import Animations from '@renderer/shared/ui/Animation/Data';
+import { WalletType } from '@renderer/shared/core';
+
+type Props = {
+  isOpen: boolean;
+  size?: number;
+  onClose: () => void;
+  onComplete: () => void;
+};
+
+const qrCode = new QRCodeStyling(NWQRConfig);
+
+export const NovaWallet = ({ isOpen, onClose, onComplete }: Props) => {
+  const { t } = useI18n();
+  const { showStatus } = useStatusContext();
+
+  const session = useUnit(wcModel.$session);
+  const client = useUnit(wcModel.$client);
+  const pairings = useUnit(wcModel.$pairings);
+  const uri = useUnit(wcModel.$uri);
+  const connect = useUnit(wcModel.events.connect);
+  const disconnect = useUnit(wcModel.events.disconnect);
+  const step = useUnit(wcOnboardingModel.$step);
+  const startOnboarding = useUnit(wcOnboardingModel.events.startOnboarding);
+
+  const previousPairings = usePrevious(pairings);
+
+  const [pairingTopic, setPairingTopic] = useState<string>();
+
+  const ref = useRef(null);
+
+  useEffect(() => {
+    if (ref.current) {
+      qrCode.append(ref.current);
+    }
+  }, []);
+
+  useEffect(() => {
+    qrCode.update({
+      data: uri,
+    });
+  }, [uri]);
+
+  useEffect(() => {
+    let timeout: any;
+    if (isOpen) {
+      startOnboarding();
+
+      timeout = setTimeout(onClose, 5 * 60 * 1000);
```

Comment:
we should move that to some const variable
https://github.com/novasamatech/nova-spektr/pull/1033/files#r1358218379